File: Livro - Módulo Java/4 - Estrutura de Repetição/Exercicio39.java
// Repita o exercício anterior para um número indefinido de bois

import java.util.Scanner;

public class Exercicio39 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int numeroBoi = 1;
        int boiMaisGordo = 0;
        int boiMaisMagro = 0;
        double pesoMaisGordo = 0;
        double pesoMaisMagro = Double.MAX_VALUE;
        double somaPesos = 0;

        while (true) {
            System.out.print("Digite o peso do boi #" + numeroBoi + " (ou digite 0 para encerrar): ");
            double pesoBoi = scanner.nextDouble();

            if (pesoBoi == 0) {
                break;
            }

            somaPesos += pesoBoi;

            if (pesoBoi > pesoMaisGordo) {
                pesoMaisGordo = pesoBoi;
                boiMaisGordo = numeroBoi;
            }

            if (pesoBoi < pesoMaisMagro) {
                pesoMaisMagro = pesoBoi;
                boiMaisMagro = numeroBoi;
            }

            numeroBoi++;
        }

        if (numeroBoi == 1) {
            System.out.println("Nenhum boi registrado.");
        } else {
            double mediaPesos = somaPesos / (numeroBoi - 1);

            System.out.println("\nResultados para um número indefinido de bois:");
            System.out.println("Boi mais gordo: #" + boiMaisGordo + " com peso " + pesoMaisGordo);
            System.out.println("Boi mais magro: #" + boiMaisMagro + " com peso " + pesoMaisMagro);
            System.out.println("Média de pesos: " + mediaPesos);
        }

        scanner.close();
    }
}

File: Livro - Módulo Python/4 - Estrutura de Repetição/Exercicio.39.py
#39. Repita o exercício anterior para um número indefinido de bois. 

soma = 0
numerosmenorpeso = 1
numerosmaiorpeso = 1
pesoboi = 0
menorpeso = 0
maiorpeso = 0
contador = 0

quantidadeBois = int(input('Digite a quantidade de bois: '))

for x in range (quantidadeBois):
    pesoboi = int(input("Digite o peso do boi: "))
    if contador == 0:
        menorpeso = maiorpeso = pesoboi
    else:
        if menorpeso == pesoboi:
            numerosmenorpeso += 1
        if menorpeso > pesoboi:
            menorpeso = pesoboi
            numerosmenorpeso = 1
        if maiorpeso == pesoboi:
            numerosmaiorpeso += 1
        if maiorpeso < pesoboi:
            maiorpeso = pesoboi
            numerosmaiorpeso = 1

    contador += 1
    soma += pesoboi

media = soma / contador
print(f'Menor peso: {menorpeso}. Existem {numerosmenorpeso} boi(s) com este peso')
print(f'Maior peso: {maiorpeso}. Existem {numerosmaiorpeso} boi(s) com este peso')
print('A média entre a massa dos bois é de', media)